### ISSUE TITLE
hint and tooltip for tabs - no translation, removes linebreaks

### ISSUE THREAD

**Issue by bwl21** [vitmalina/w2ui#1322]
I have a Tab with text and tooltip.

* Text is not translated
* tooltip is translated, but does not follow line breaks

**Comment by bwl21** [comment]
@vitmalina Hi, I fixes the first issue. The second one seems to be a deeper issue. Could you please have a look on this?

**Comment by mrg2001** [comment]
Can you please provide a screenshort of tooltip on Menus (I am not sure what you mean by Menus). Below is screenshot for tooltip on tabs.

<img width="223" alt="screen shot 2016-08-30 at 7 14 28 am" src="https://cloud.githubusercontent.com/assets/1964803/18092570/a27a853a-6e81-11e6-87b5-12d679d64d6b.png">

The same tooltip is also implemented on toolbar and grid columns. I have also added w2utils.lang() for text on the tab.

As for second issue, can you provide some code that is not working.

**Comment by bwl21** [comment]
This is the tooltip on a menu (implemented by a title attribute)

![screenshot_651](https://cloud.githubusercontent.com/assets/2022976/18092987/aedb27c4-6ece-11e6-9c88-27d631a710cd.jpg)

This is the same tooltip on a tab (implemented by jquery)

![screenshot_652](https://cloud.githubusercontent.com/assets/2022976/18092998/b9e11232-6ece-11e6-8efa-64f28cf86e24.jpg)

I would not care that much about the design of the tooltip. But it should respect newlines.

**Comment by mrg2001** [comment]
I see. Yes, you can still use title attribute in HTML. It should not be a problem. You can even use it in tab. You just need to wrap tab caption in a div. Then it would work the same way. However, as you know title attribute does not support html formatting. It does support new lines, I believe.

As for translation, I do not know how it removes new lines as it is quite straightforward. I call w2utils.lang("you title") and it looks through w2util.settings.phrases object and if there is an exact match, then it will replace it. It does not replace partial text and does no processing. 

Besides I think that the reason new line codes are not working because HTML ignores them. And tooltip is just a div with HTML in it. You need to replace \n with <br> and it will work.

**Comment by bwl21** [comment]
Its not me using title attribute. it is w2ui transferring the tooltip of a menu item to the title attribute while it creates HTML on other places.

The translation does not do anything with with new lines. So you are right the reason that nes line codes are not working is, because tabs and form fields use w2tag to create the tooltip.

So I eventually propose:

1. use w2tag to create the tooltip even for toolbar items and toolbar/menu items. By this all tooltips look the same.

2. let w2tag check if it gets a html fragment (if it starts with a tag opening character)

  * if it is html - provide this
  * otherwise replace \n by <br/>

**Comment by mrg2001** [comment]
Yes, I see that menu item tooltip is rendered as title. I will make a note to refactor it. As for replacing \n, is it a problem for you to do it before you set tooltip?

**Comment by bwl21** [comment]
To be honest, i would consider it as a cool feature in general if w2tag would recognize if it has plain text or html. It would be a lot of effort to do it before setting tooltip. In particular, I also have to change all the translation keys etc.
